### PR TITLE
Ocean: block indexing

### PR DESCRIPTION
**Pull request by canonbrother**
## Summary

- add `--experimental_allow_proto3_optional` 
- ffi passing pindex into ocean_index_block
- indexing blocks
- update block apis to grab indexed block

#### TODO

- add mn owner address and operator address to block
- handle optional `next` query on list block

###### Get `/blocks/a9fecf3aaa8e9c103b1fcde34d1b37c55b1b4efc0ff47f7ddd7695a1d2480856`

```
curl -X GET -H 'content-type:application/json' http://0.0.0.0:3002/blocks/a9fecf3aaa8e9c103b1fcde34d1b37c55b1b4efc0ff47f7ddd7695a1d2480856

{"id":"a9fecf3aaa8e9c103b1fcde34d1b37c55b1b4efc0ff47f7ddd7695a1d2480856","hash":"a9fecf3aaa8e9c103b1fcde34d1b37c55b1b4efc0ff47f7ddd7695a1d2480856","previousHash":"134f09f047f34a18e9b979122e57aead2c82818ce7737388234c8d4ae9f9949a","height":126,"version":536870912,"time":1704930017,"medianTime":1704930015,"transactionCount":2,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":126,"stakeModifier":"5f1686f613ae7d04952ccd84aefedc59a71d22c61813d80d1da89603e2f09073","merkleroot":"15ae8a6966ddb5ef9213cb4a073ed7b43247a5308c9424f1e7da34675d64f1cd","size":1317,"sizeStripped":1281,"weight":5160}%
```

###### List `/blocks?size=30`
```
curl -X GET -H 'content-type:application/json' http://0.0.0.0:3002/blocks\?size\=30
{"data":[{"id":"ba7bf591494171054355931111a93f1097b4cd0995369506b6e09a9982308a09","hash":"ba7bf591494171054355931111a93f1097b4cd0995369506b6e09a9982308a09","previousHash":"0091f00915b263d08eba2091ba70ba40cea75242b3f51ea29f4a1b8d7814cd01","height":1,"version":536870912,"time":1704933158,"medianTime":1704933158,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":1,"stakeModifier":"fdd82eafa32300653d3b2d9b98a6650b4b15fe2eb32cdd847d3bf2272514cfbf","merkleroot":"00eed320c213f506038fa29f77d4d2535232fa97b7789ff6fb516c63201c5e44","size":396,"sizeStripped":360,"weight":1476},{"id":"6b4902a516d9d8570441e91a269bf0f33397193d7f85a009ed0a44a8086ffbcb","hash":"6b4902a516d9d8570441e91a269bf0f33397193d7f85a009ed0a44a8086ffbcb","previousHash":"ba7bf591494171054355931111a93f1097b4cd0995369506b6e09a9982308a09","height":2,"version":536870912,"time":1704933160,"medianTime":1704933158,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":2,"stakeModifier":"cc65dfe3cb4ceca7fc1ff0eb266499426ddb2ba8d4ed0b6c5625bfb707f7254a","merkleroot":"a95a4e99ac7c734de8ec543f746d611ef8974ba76f3b384d8ed7f487d5e99b55","size":396,"sizeStripped":360,"weight":1476},{"id":"296f7a8adb6fe05a2034f563ad198873264dda31293a56c2344de8c373eccfa2","hash":"296f7a8adb6fe05a2034f563ad198873264dda31293a56c2344de8c373eccfa2","previousHash":"6b4902a516d9d8570441e91a269bf0f33397193d7f85a009ed0a44a8086ffbcb","height":3,"version":536870912,"time":1704933160,"medianTime":1704933160,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":3,"stakeModifier":"f1042a3fc047350eed4a44bf83485c0024f8f16f4b7ec49cfbaa62f839dcffe2","merkleroot":"17098d19ee9c10d6a766ba55169d2c3a99bddaf5a675af7474108f5681fc6bd0","size":396,"sizeStripped":360,"weight":1476},{"id":"198a677ad90aede43de008135eb1eccc3bfe8e6cea5590c47627df3494ac5452","hash":"198a677ad90aede43de008135eb1eccc3bfe8e6cea5590c47627df3494ac5452","previousHash":"296f7a8adb6fe05a2034f563ad198873264dda31293a56c2344de8c373eccfa2","height":4,"version":536870912,"time":1704933162,"medianTime":1704933160,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":4,"stakeModifier":"b9437afd8eca3f82feca5d72e4f70c462bfa2f833dff9d2aedd99db697c0bc94","merkleroot":"1db567134cbbfeaaedb6090e7d26d15aa940871441494117664dc54d171b03b7","size":396,"sizeStripped":360,"weight":1476},{"id":"7bb0169a5b0e15ad5a8041c8343bc2f99071aed294aa6ebaad02971da7329e7b","hash":"7bb0169a5b0e15ad5a8041c8343bc2f99071aed294aa6ebaad02971da7329e7b","previousHash":"198a677ad90aede43de008135eb1eccc3bfe8e6cea5590c47627df3494ac5452","height":5,"version":536870912,"time":1704933162,"medianTime":1704933160,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":5,"stakeModifier":"f0cda7c8dfab7a90b6acd2f79a66e603e999a52dfa4a8e3c821e842e39202f45","merkleroot":"5a7efb3034f7ae0c7e1068b096b0b688e9e8e48e483371bb23238edadf295aff","size":396,"sizeStripped":360,"weight":1476},{"id":"36472ae3a7397a6233902430a1552f71eef2295548dc395bd8702d412513c3a8","hash":"36472ae3a7397a6233902430a1552f71eef2295548dc395bd8702d412513c3a8","previousHash":"7bb0169a5b0e15ad5a8041c8343bc2f99071aed294aa6ebaad02971da7329e7b","height":6,"version":536870912,"time":1704933162,"medianTime":1704933160,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":6,"stakeModifier":"80ef191722380bb3479fa8d57d9871ddcdc723f7059722851aa3747d21aba585","merkleroot":"e76b47b4ec463962aa7e175a3d02a2c1c5a3e9ee7026d03d695767967ee33859","size":396,"sizeStripped":360,"weight":1476},{"id":"c453680a4d3e9fe124699878fbd1f2df45141c9f6717c6087364044d3be5fd4c","hash":"c453680a4d3e9fe124699878fbd1f2df45141c9f6717c6087364044d3be5fd4c","previousHash":"36472ae3a7397a6233902430a1552f71eef2295548dc395bd8702d412513c3a8","height":7,"version":536870912,"time":1704933162,"medianTime":1704933162,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":7,"stakeModifier":"d9d7a3c6a065f185adffdb525b9c5d6f31ac731d5f4894f79a2168c2a9edc7aa","merkleroot":"2bde0e08467c87d9ebdcb8dbc360c8d5ead19a4f4dac473c177d9828e269b015","size":396,"sizeStripped":360,"weight":1476},{"id":"ea1767e3a9197cc2e9dd4a4042018c3990eeeb41fefd26149863f586c2d899e7","hash":"ea1767e3a9197cc2e9dd4a4042018c3990eeeb41fefd26149863f586c2d899e7","previousHash":"c453680a4d3e9fe124699878fbd1f2df45141c9f6717c6087364044d3be5fd4c","height":8,"version":536870912,"time":1704933164,"medianTime":1704933162,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":8,"stakeModifier":"2f49395c7c3e70ffb0d9a1a03a5278c6e16d220287483b79b30c1a4ac55b6779","merkleroot":"109eaf1fef1101e1bced75c237dd15718b4a00fc15ccbe8ca9ae44e34989d200","size":396,"sizeStripped":360,"weight":1476},{"id":"177ff5650474bc7bcf89abe8cfa8eb112e3c9927e9306c5b9fac7901be0fad5c","hash":"177ff5650474bc7bcf89abe8cfa8eb112e3c9927e9306c5b9fac7901be0fad5c","previousHash":"ea1767e3a9197cc2e9dd4a4042018c3990eeeb41fefd26149863f586c2d899e7","height":9,"version":536870912,"time":1704933164,"medianTime":1704933162,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":9,"stakeModifier":"2ebf24d36051d9eda80af78956b69f89b21e18f74cc229daeaafe9aa3013eb7f","merkleroot":"1fd688498d75035c1ba5d013097b6351295db1fce5f83cd9c99d516dee4368ac","size":396,"sizeStripped":360,"weight":1476},{"id":"db2db28af51db83b079a680619a3f01af2e20274a900763e9ed78140bafbd60e","hash":"db2db28af51db83b079a680619a3f01af2e20274a900763e9ed78140bafbd60e","previousHash":"177ff5650474bc7bcf89abe8cfa8eb112e3c9927e9306c5b9fac7901be0fad5c","height":10,"version":536870912,"time":1704933164,"medianTime":1704933164,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":10,"stakeModifier":"51002418ebf0f4ccbb892ac7f7e72d2323afd4b480402c78c998373ac155b127","merkleroot":"85c253658946df4545cee786c935c06290856c7bcee024508256b5e08d2e4c46","size":396,"sizeStripped":360,"weight":1476},{"id":"a7626c27b93a49a47866b11f0ed03adb576c6b28da0270a44807bb2373fe555f","hash":"a7626c27b93a49a47866b11f0ed03adb576c6b28da0270a44807bb2373fe555f","previousHash":"db2db28af51db83b079a680619a3f01af2e20274a900763e9ed78140bafbd60e","height":11,"version":536870912,"time":1704933166,"medianTime":1704933164,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":11,"stakeModifier":"f70d9c80e26129684351bdf941bb270cc94dd5d298339d2574e007c80d469957","merkleroot":"5b48863eccfe1da1aef911f4cdd12b79e7ac6ab1069e422550c3860b4f9e2fe6","size":396,"sizeStripped":360,"weight":1476},{"id":"9181d814429917d6321f71ca4a6a6e7c5cd5ecc8f885b5f943838be123c9f885","hash":"9181d814429917d6321f71ca4a6a6e7c5cd5ecc8f885b5f943838be123c9f885","previousHash":"a7626c27b93a49a47866b11f0ed03adb576c6b28da0270a44807bb2373fe555f","height":12,"version":536870912,"time":1704933166,"medianTime":1704933164,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":12,"stakeModifier":"ea008db022e19e1fb16e7987c8b358b750590b3c5fa156bc7055263c4eff438c","merkleroot":"56831ddd2a36856484fa4b78cfe4218be04083573f60739edc1b3e35e97a08d8","size":396,"sizeStripped":360,"weight":1476},{"id":"a0d3efad771da8e53ef180ae21aa0239808a2003ec7c2076d79410c8dd0a7042","hash":"a0d3efad771da8e53ef180ae21aa0239808a2003ec7c2076d79410c8dd0a7042","previousHash":"9181d814429917d6321f71ca4a6a6e7c5cd5ecc8f885b5f943838be123c9f885","height":13,"version":536870912,"time":1704933166,"medianTime":1704933166,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":13,"stakeModifier":"8667cbbc355da8734f0ee05ba41fc05c08837cc74d2616d83523abaca1449f97","merkleroot":"11690b8a9948119692f316f24c8e1418f69989e4ccea767f80c5b277ea6948e9","size":396,"sizeStripped":360,"weight":1476},{"id":"89047cd7cc2ae933a1eaee17a1fb32dc86691dfe70c9a0f3c0d0b6e9b8a50beb","hash":"89047cd7cc2ae933a1eaee17a1fb32dc86691dfe70c9a0f3c0d0b6e9b8a50beb","previousHash":"a0d3efad771da8e53ef180ae21aa0239808a2003ec7c2076d79410c8dd0a7042","height":14,"version":536870912,"time":1704933168,"medianTime":1704933166,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":14,"stakeModifier":"282c3bb9a67661684dd8d5909dc10b6ebbb1aabdfac3a514a95a949b4c66df66","merkleroot":"1c4b2ede71f219f1c8976a3e84610f5a805cf46be3491668428681a11c8e9196","size":396,"sizeStripped":360,"weight":1476},{"id":"4e7f266de2dc6440bcca6584547705de8ed05b27dfe57cc5df252eb3bc750ef9","hash":"4e7f266de2dc6440bcca6584547705de8ed05b27dfe57cc5df252eb3bc750ef9","previousHash":"89047cd7cc2ae933a1eaee17a1fb32dc86691dfe70c9a0f3c0d0b6e9b8a50beb","height":15,"version":536870912,"time":1704933168,"medianTime":1704933166,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":15,"stakeModifier":"5702d841b0200530feaecead3fab0e9058044a405a32ba78163f743b0f48de3a","merkleroot":"b502fc3eec458f900bf598a449ad772e2259e1dc8a9c7b5ad0e5f66857f97006","size":396,"sizeStripped":360,"weight":1476},{"id":"1e07b5184d9b219f62079a9237fc6b7c89432a399310acb0a384e427229db6b4","hash":"1e07b5184d9b219f62079a9237fc6b7c89432a399310acb0a384e427229db6b4","previousHash":"4e7f266de2dc6440bcca6584547705de8ed05b27dfe57cc5df252eb3bc750ef9","height":16,"version":536870912,"time":1704933168,"medianTime":1704933168,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":16,"stakeModifier":"73299455a8b3e64959ef401f35872095ea443f8a346961a3b471692984b30f2e","merkleroot":"a5f805ab908185741254e9cfd39d08607290fa5653c57ea3811f1dd447dbb892","size":363,"sizeStripped":327,"weight":1344},{"id":"fdee34728e02169a93daad077ce3caa4c7a336cbc8601257d8aea5010cbee928","hash":"fdee34728e02169a93daad077ce3caa4c7a336cbc8601257d8aea5010cbee928","previousHash":"1e07b5184d9b219f62079a9237fc6b7c89432a399310acb0a384e427229db6b4","height":17,"version":536870912,"time":1704933170,"medianTime":1704933168,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":17,"stakeModifier":"b9294ee91cd7aa4d41b768ec47e6f0685b94b29dad74c7412959a575e691b22c","merkleroot":"56ae8f861d3739b3fb49589d161e3c957aba8980fa3bd0660b2939b440b70a01","size":364,"sizeStripped":328,"weight":1348},{"id":"a03bb6f9fcc1f22dc5d47f874cf4af62eb0ac076418f11b35a097d9f0e8f8582","hash":"a03bb6f9fcc1f22dc5d47f874cf4af62eb0ac076418f11b35a097d9f0e8f8582","previousHash":"fdee34728e02169a93daad077ce3caa4c7a336cbc8601257d8aea5010cbee928","height":18,"version":536870912,"time":1704933170,"medianTime":1704933168,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":18,"stakeModifier":"a02c98218f7be4ff0a5726ce76b8d60dfaad5cce880af34b6d3d9b224f9d1d52","merkleroot":"2c9cecdcea4fa9aeff9ed109112d6c56c3683bbf99e03bb03c15f20030ea1fee","size":364,"sizeStripped":328,"weight":1348},{"id":"6cf24c3bff91d83a5e0662f1b135b6dd6508dec81255e4dd920ebd296a1c67be","hash":"6cf24c3bff91d83a5e0662f1b135b6dd6508dec81255e4dd920ebd296a1c67be","previousHash":"a03bb6f9fcc1f22dc5d47f874cf4af62eb0ac076418f11b35a097d9f0e8f8582","height":19,"version":536870912,"time":1704933170,"medianTime":1704933170,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":19,"stakeModifier":"b989eb7b8fa32ca0d8dd8fee6b6bf9bcc1443241cf25528a6acac2bf4f023f34","merkleroot":"b6e506f6788dc68b84630c8598ae6cdbb1ad404494fdec4bded0fb6111503704","size":364,"sizeStripped":328,"weight":1348},{"id":"17f05452ffc73c5148afad1e29ac000ea513ef3a781236304950f51d53dc7130","hash":"17f05452ffc73c5148afad1e29ac000ea513ef3a781236304950f51d53dc7130","previousHash":"6cf24c3bff91d83a5e0662f1b135b6dd6508dec81255e4dd920ebd296a1c67be","height":20,"version":536870912,"time":1704933172,"medianTime":1704933170,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":20,"stakeModifier":"b108433e387f685283c540710d6cd248e5328d42cbc1983243e1702ff00fd17f","merkleroot":"7ae9f7ab542a2075be248d91a1f8fb454c23fa6b1b5bc84a74b99c8f32757394","size":364,"sizeStripped":328,"weight":1348},{"id":"5bffea81473eca03b899d92f0dc7e8983f76faa6550011d7e89b93f80775075d","hash":"5bffea81473eca03b899d92f0dc7e8983f76faa6550011d7e89b93f80775075d","previousHash":"17f05452ffc73c5148afad1e29ac000ea513ef3a781236304950f51d53dc7130","height":21,"version":536870912,"time":1704933172,"medianTime":1704933170,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":21,"stakeModifier":"a1f7d1f97e1ddd8cefe66d12a6800d4e2ca59078634192e2e834a2b46ba63a92","merkleroot":"4c3d5c240d971a90a223563eb931c4385ecf9168f25f532b90884020fa404af6","size":364,"sizeStripped":328,"weight":1348},{"id":"64361090af630b6034b4bfe53ca9fe300f6ce470f0212984108075e7ee3fb8a4","hash":"64361090af630b6034b4bfe53ca9fe300f6ce470f0212984108075e7ee3fb8a4","previousHash":"5bffea81473eca03b899d92f0dc7e8983f76faa6550011d7e89b93f80775075d","height":22,"version":536870912,"time":1704933172,"medianTime":1704933172,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":22,"stakeModifier":"7caef77a67eefbf99a7eb627dba0e78653564c227238d38ca226679722693593","merkleroot":"3e9dfda8996c2a81271416ecf8f7946ca0f6d695162142be441e6d089c8409f3","size":364,"sizeStripped":328,"weight":1348},{"id":"4cd38e1840b3310b259d12b5def5e121f898fed6e111340390e0bf9d046e1c1e","hash":"4cd38e1840b3310b259d12b5def5e121f898fed6e111340390e0bf9d046e1c1e","previousHash":"64361090af630b6034b4bfe53ca9fe300f6ce470f0212984108075e7ee3fb8a4","height":23,"version":536870912,"time":1704933174,"medianTime":1704933172,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":23,"stakeModifier":"ce988f60c3a43ee419d656156a667b8ccc590541363628ade54fa27be3f81690","merkleroot":"f7900f3d9afe3959c56c4eddb0533decba226f6eb0db4bf1579cc759fdef9fb8","size":364,"sizeStripped":328,"weight":1348},{"id":"c1672d2bacbab8df14785693c7471d51582d693c9a60908a729c0591c79decae","hash":"c1672d2bacbab8df14785693c7471d51582d693c9a60908a729c0591c79decae","previousHash":"4cd38e1840b3310b259d12b5def5e121f898fed6e111340390e0bf9d046e1c1e","height":24,"version":536870912,"time":1704933174,"medianTime":1704933172,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":24,"stakeModifier":"71268fe18d15b052639c1a389b0daa6acb724f33871f1cd0852964861255cbf5","merkleroot":"1fc61d0c8047eaac7abed358a10cc66d27dda1f0597d84141430a3a8b5e438cb","size":364,"sizeStripped":328,"weight":1348},{"id":"d714cc8fb1aefbfbf570ca85c463cd31711bdd36367f77b69ff7895da96f38e8","hash":"d714cc8fb1aefbfbf570ca85c463cd31711bdd36367f77b69ff7895da96f38e8","previousHash":"c1672d2bacbab8df14785693c7471d51582d693c9a60908a729c0591c79decae","height":25,"version":536870912,"time":1704933174,"medianTime":1704933174,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":25,"stakeModifier":"f1658faecb962b3e5a837ca592030795629ff3093712135e810fc977180800b7","merkleroot":"ae5330b9c126cd961c432d312df5788e7238f8d3c01e0a30a303a432b6681b0f","size":364,"sizeStripped":328,"weight":1348},{"id":"6c5aaa14328ac63aefb94dd7e6b266b44827822ed9cf3bd20b667558595fb56b","hash":"6c5aaa14328ac63aefb94dd7e6b266b44827822ed9cf3bd20b667558595fb56b","previousHash":"d714cc8fb1aefbfbf570ca85c463cd31711bdd36367f77b69ff7895da96f38e8","height":26,"version":536870912,"time":1704933176,"medianTime":1704933174,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":26,"stakeModifier":"a3299227f6a5f8652690cb2b6f5cb6fecdb9f47ee278386c39374375bd7b2e48","merkleroot":"47a0a1519df0d7a31f2122eedc16d7eb502af71bc9e6dee97348896bc0764f67","size":364,"sizeStripped":328,"weight":1348},{"id":"05df73a95170ca593fb81fb4dbb0ee0d268411df5d66cd888c865ac91b7f631e","hash":"05df73a95170ca593fb81fb4dbb0ee0d268411df5d66cd888c865ac91b7f631e","previousHash":"6c5aaa14328ac63aefb94dd7e6b266b44827822ed9cf3bd20b667558595fb56b","height":27,"version":536870912,"time":1704933176,"medianTime":1704933174,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":27,"stakeModifier":"c2b8452683b689a4fcff80bbe39c5bf7a9e245389d349956824027216cead6ba","merkleroot":"a13385508676289b92f2409c8fb15271261e8d3e4c64c20fd49ad9f54549c148","size":364,"sizeStripped":328,"weight":1348},{"id":"1601aed1960fc559d6d2b26e05bfd4cae4ca70db332f17f1d50ec6ea92d78b3a","hash":"1601aed1960fc559d6d2b26e05bfd4cae4ca70db332f17f1d50ec6ea92d78b3a","previousHash":"05df73a95170ca593fb81fb4dbb0ee0d268411df5d66cd888c865ac91b7f631e","height":28,"version":536870912,"time":1704933176,"medianTime":1704933176,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":28,"stakeModifier":"833cf42033286abad86de3c78e25200651d1f178eb7a35339facc5b38ab29590","merkleroot":"93d147072de7196542b5e8aed2d5a3b25ed8a801b17b92bc1d02844dffa618a8","size":364,"sizeStripped":328,"weight":1348},{"id":"7302709f20c3087806384177162792c99146416c48b38cb504882a899531f97b","hash":"7302709f20c3087806384177162792c99146416c48b38cb504882a899531f97b","previousHash":"1601aed1960fc559d6d2b26e05bfd4cae4ca70db332f17f1d50ec6ea92d78b3a","height":29,"version":536870912,"time":1704933178,"medianTime":1704933176,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":29,"stakeModifier":"0779d13b698097f172d69f5d14c2615a4feee796c06c03a46722884a55f689d6","merkleroot":"d6ccf9950ce75ef463f79a20995aac337d22ba37bb657ee9dbf5737a670278ea","size":364,"sizeStripped":328,"weight":1348},{"id":"716ab64b38c03a637045004150ff190e3db45b29f62e1e7e7c0875b797bd5fe6","hash":"716ab64b38c03a637045004150ff190e3db45b29f62e1e7e7c0875b797bd5fe6","previousHash":"7302709f20c3087806384177162792c99146416c48b38cb504882a899531f97b","height":30,"version":536870912,"time":1704933178,"medianTime":1704933176,"transactionCount":1,"difficulty":545259519,"masternode":"","minter":"","minterBlockCount":30,"stakeModifier":"f2b1e7c270364f15a4494780e2dba0cf94516f79228de09d3b45b94af5cad6e8","merkleroot":"6f2359bbf7daa899116a2a5ab21d5439b2d12b16b55ac1e74f1826f0d5592e10","size":364,"sizeStripped":328,"weight":1348}],"page":{"next":"716ab64b38c03a637045004150ff190e3db45b29f62e1e7e7c0875b797bd5fe6"}}%
```

### CODE REVIEW

File: lib/ain-grpc/build.rs [hunk 1]
```diff
@@ -266,6 +266,7 @@ fn compile_proto_and_generate_services(
         // There's no way to compile protos using custom generator in tonic,
         // so we're left with creating a prost config and using that for codegen.
         let mut config = Config::new();
+        config.protoc_arg("--experimental_allow_proto3_optional");
```
Comment (Jouzo) [inline lib/ain-grpc/build.rs:269]: Do we need this here ?

**Review by Jouzo — APPROVED** [review]

File: lib/ain-grpc/build.rs (hunk 1 above)
Comment (canonbrother) [reply lib/ain-grpc/build.rs:269]: yea.. i got asked to add the flag 😕 
```
Caused by:
  process didn't exit successfully: `/home/canonbrother/dev/DeFiCh/ain/build/lib/target/debug/build/ain-grpc-3d03a453500eafb3/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=/home/canonbrother/dev/DeFiCh/ain/lib/proto/types/tx.proto
  cargo:rerun-if-changed=/home/canonbrother/dev/DeFiCh/ain/lib/proto/types/eth.proto
  cargo:rerun-if-changed=/home/canonbrother/dev/DeFiCh/ain/lib/proto/types/block.proto

  --- stderr
  thread 'main' panicked at 'compiling protobuf: Custom { kind: Other, error: "protoc failed: types/eth.proto: This file contains proto3 optional fields, but --experimental_allow_proto3_optional was not set.\n" }', ain-grpc/build.rs:274:14
'``

Comment (canonbrother) [reply lib/ain-grpc/build.rs:269]: hmm.. wait.. seems this's blocking macos

File: lib/ain-ocean/src/model/block.rs [hunk 2]
```diff
@@ -7,21 +7,20 @@ pub struct Block {
     pub id: String,
     pub hash: String,
     pub previous_hash: String,
```
Comment (Jouzo) [inline lib/ain-ocean/src/model/block.rs:9]: Can use BlockHash here

**Pull request merged**